src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,977 @@
+                ExpressionSyntax initializer,
+                string variableName,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                var parentStatement = invocationExpression.GetAncestorOrThis<StatementSyntax>();
+                if (parentStatement != null)
+                {
+                    if (TryConvertIfInInvocationInternal(
+                        invocationExpression,
+                        typeCheckMethod,
+                        parentStatement,
+                        initializer,
+                        variableName,
+                        out var variable,
+                        out var nodesBefore,
+                        out var nodesAfter))
+                    {
+                        var statements = GenerateStatements(expression => leafExpressionCreationMethod(variable, expression), queryExpressionProcessingInfo);
+                        var list = new List<CSharpSyntaxNode>();
+                        list.AddRange(nodesBefore);
+                        list.AddRange(statements);
+                        list.AddRange(nodesAfter);
+                        documentUpdateInfo = new DocumentUpdateInfo(parentStatement, list);
+                        return true;
+                    }
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInInvocationInternal(
+                InvocationExpressionSyntax invocationExpression,
+                Func<ITypeSymbol, bool> typeCheckMethod, // TODO change to SyntaxNode, bool ?
+                StatementSyntax parentStatement,
+                ExpressionSyntax initializer,
+                string variableName,
+                out ExpressionSyntax variable,
+                out CSharpSyntaxNode[] nodesBefore,
+                out CSharpSyntaxNode[] nodesAfter)
+            {
+                var invocationParent = invocationExpression.Parent;
+                var symbolName = GetFreeSymbolNameAndMarkUsed(variableName);
+                switch (invocationParent.Kind())
+                {
+                    case SyntaxKind.EqualsValueClause:
+                        // Avoid for(int i = (from x in a select x).Count(); i < 10; i++)
+                        if (invocationParent.IsParentKind(SyntaxKind.VariableDeclarator, SyntaxKind.VariableDeclaration, SyntaxKind.LocalDeclarationStatement) &&
+                            // Avoid int i = (from x in a select x).Count(), j = i;
+                            ((VariableDeclarationSyntax)invocationParent.Parent.Parent).Variables.Count == 1)
+                        {
+                            var variableDeclarator = ((VariableDeclaratorSyntax)invocationParent.Parent);
+                            if (typeCheckMethod(_semanticModel.GetTypeInfo(((VariableDeclarationSyntax)variableDeclarator.Parent).Type).Type))
```

Comment:
>_semanticModel.GetTypeInfo(((VariableDeclarationSyntax)variableDeclarator.Parent).Type).Type) [](start = 48, length = 93)

I would GetDeclaredSymbol and get type from that local instead, but you don't have to change it in this PR.

Reply:
Or even at all.

---
In reply to: [178422896](https://github.com/dotnet/roslyn/pull/25362#discussion_r178422896) [](ancestors = 178422896)